src/libexpr/parser.y

```diff
@@ -497,7 +501,7 @@ attr
 
 string_attr
   : '"' string_parts '"' { $$ = $2; }
-  | DOLLAR_CURLY expr '}' { $$ = $2; }
+  | DOLLAR_CURLY expr '}' { $$ = new std::variant<Expr *, std::string_view>($2); }
```

Comment:
I'm especially fond of the `std::variant` :)

Any reasons we need the `new` here? Seems like a fine fixed-size thing so we have the opportunity to stop leaking memory.

Reply:
Ah this is mutation/assignment not definition.

https://en.cppreference.com/w/cpp/utility/variant/variant.html

So the no argument constructor wants to make the first variant. Try switching the order so the `std::string_view` is first. That one is definitely default-constructable.

Reply:
I'm pretty sure pointers are also trivially default-constructible. In any case I remember tried it both ways, it didn't work either way. I can try it again and pull up the exact error if you want, but I'm off to bed so it'll have to be tomorrow.